tests/test_publisher.py

```diff
@@ -90,24 +137,64 @@ def test_publish_to_invalid_destination(connection: Connection) -> None:
     assert raised is True
 
 
+def test_publish_per_message_to_invalid_destination(connection: Connection) -> None:
+
+    queue_name = "test-queue-1"
+    raised = False
+
+    message = Message(body="test")
+    message = AddressHelper.message_to_address_helper(
+        message, "/invalid_destination/" + queue_name
+    )
+    publisher = connection.publisher()
+
+    try:
+        publisher.publish(message)
+    except ArgumentOutOfRangeException:
+        raised = True
+    except Exception:
+        raised = False
+
+    if publisher is not None:
+        publisher.close()
+
+    assert raised is True
+
+
+def test_publish_per_message_both_address(connection: Connection) -> None:
+
+    queue_name = "test-queue-1"
+    raised = False
+
+    management = connection.management()
+    management.declare_queue(QuorumQueueSpecification(name=queue_name))
+
+    message = Message(body="test")
+    message = AddressHelper.message_to_address_helper(message, "/queues/" + queue_name)
```

Comment:
Can you move `message = AddressHelper.message_to_address_helper(message, "/queues/" + queue_name)` down, close to the publisher? and use the helpers instead of `"/queues/" + queue_name`